readalongs/util.py

```diff
@@ -1,10 +1,22 @@
-import g2p.mappings.langs as g2p_langs
-from networkx import has_path
+import re
+from collections.abc import Iterable
+from itertools import tee
+
+import click
 
 LANGS = None
 LANG_NAMES = None
 
 
+def getLangsDeferred() -> Iterable:
+    """Lazilly get the list of language codes supported by g2p library
+
+    Yields an Iterable in such a way that the g2p database is only loaded when
+    the results are iterated over, rather than when this function is called.
+    """
+    yield from getLangs()[0]
```

Comment:
Oh, this is cool!

Reply:
Yeah, generators in Python are strictly lazily evaluated. I tested, and if there's a yield statement in the function, nothing at all in the function runs until `next()` is first called on it.